packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -656,14 +657,22 @@ class _RangeSliderState extends State<RangeSlider> with TickerProviderStateMixin
           theme.colorScheme.primary.withOpacity(0.12);
     }
 
+    final Color effectiveActiveTrackColor =
+        widget.activeColor ?? sliderTheme.activeTrackColor ?? theme.colorScheme.primary;
+    final Color effectiveInactiveTrackColor =
+        widget.inactiveColor ??
+        sliderTheme.inactiveTrackColor ??
+        theme.colorScheme.primary.withOpacity(0.24);
+    final double effectiveTrackHeight =
+        effectiveActiveTrackColor != Colors.transparent &&
```

Comment:
In the other Slider fix PR(#161814), it seems only when both track colors are transparent, we set the height to 0. But here it looks like as long as one of the track colors is transparent, we set the height to 0?

Reply:
Here it checks if both colors are transparent with a `&&` condition, same as https://github.com/flutter/flutter/pull/161814
`effectiveActiveTrackColor != Colors.transparent && effectiveInactiveTrackColor != Colors.transparent`

Reply:
it checks if both active and inactive colors are transparent. 

I moved the check into `BaseRangeSliderTrackShape` to match with https://github.com/flutter/flutter/pull/161814. That's why I conditional look slightly different now.  It was returning height when colors isn't transparent  but in the base shape, we don't need to return regular height :)